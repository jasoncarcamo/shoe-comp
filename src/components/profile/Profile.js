import React from "react";
import UserService from "../../services/UserService";

export default class Profile extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            first_name: "",
            last_name: "",
            orders: [],
            isMounted: false,
            error: ""
        }
    }

    componentWillMount(){
        this.setState({isMounted: true});
    }

    componentDidMount(){
        if(this.state.isMounted){
            UserService.getUser()
                .then( res => {
                    this.setState(res)                
                });

            UserService.getOrders()
                .then( data => {
                    this.formatData(data)
                })
                .catch(error => {
                    this.setState({ error: error.error})
                })
        }
    }

    formatData = (data)=>{

        let orders = data;
        orders.forEach((order, index)=>{

            let formatData = order.items.split("");

            if(formatData[0] === "{" && formatData[formatData.length - 1] === "}"){
                formatData.shift();
                formatData.pop();
                formatData.shift();
                formatData.pop();
            };

            console.log(formatData)


            for(let i = 0; i < formatData.length; i++){
                if(formatData[i] === ("\\")){
                    formatData.splice(i, 1);                
                };
    
                if(formatData[i] === "}" && formatData[i + 1] === "\""){
                    formatData.splice(i + 1, 1);
                }
    
                if(formatData[i] === "}" && formatData[i + 2] === "\""){
                    formatData.splice(i + 2, 1);
                }
            };

            formatData = formatData.join("");

            formatData = "[" + formatData + "]";

            if(formatData === "[{}]"){
                return this.setState({ items: []});
            };

            orders[index].items = JSON.parse(formatData)

        });
        this.setState({orders})

    }

    componentWillUnmount(){
        this.setState({ isMounted: false});
    }

    renderOrders = ()=>{
        const orders = this.state.orders;

        return orders.map( (order, index) => {
            let li = [];

            for( let i = 0; i < order.items.length; i++){
                console.log(order.items[i])
                li[i] = (
                    <li key={index+"-"+ i}>
                        {order.items[i].top}
                    </li>
                )
            };
            
            return li;
        })
    }

    render(){
        console.log(this.state.orders)
        return (
            <section id="profile-section">
                <h1>Hello {this.state.first_name} {this.state.last_name}</h1>
                <ul>
                    {this.renderOrders()}
                </ul>
            </section>
        )
    }
}