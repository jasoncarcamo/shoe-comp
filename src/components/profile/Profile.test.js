import React from "react";
import ReactDOM from "react-dom";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import Profile from "./Profile";
import Enzyme, {shallow} from "enzyme";
import Adaptor from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adaptor()});


describe("<Profile/>", ()=>{
    it("Renders without crashing", ()=>{
        const div = document.createElement("div");

        ReactDOM.render(<BrowserRouter><Profile/></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Creates a snapshot", ()=>{
        const tree = renderer.create(<BrowserRouter><Profile/></BrowserRouter>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});


