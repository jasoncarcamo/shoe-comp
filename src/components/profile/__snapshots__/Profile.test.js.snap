// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> Creates a snapshot 1`] = `
<section
  id="profile-section"
>
  <h1>
    Hello 
    
     
    
  </h1>
  <p
    id="profile-no-items"
  >
    No items yet. Your first item will appear after you make you first 
    <a
      href="/shop"
      onClick={[Function]}
    >
      purchase
    </a>
    .
  </p>
</section>
`;
